services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableIslRulesSpec.groovy

```diff
@@ -0,0 +1,141 @@
+                with(islRules) { rules ->
+                    //islToManipulate.reversed it is for deleting from dstSw
+                    rules.size() == (relatedIsls[sw.dpId] - islToManipulate - islToManipulate.reversed).size()
+                    rules*.instructions.goToTable.unique() == [4]  // 4 - egress table id
+                    islRules*.match.inPort.sort() == (relatedIsls[sw.dpId] -
+                            islToManipulate - islToManipulate.reversed)*.srcPort.collect { it.toString() }.sort()
+                }
+            }
+        }
+
+        and: "The src and dst switch pass switch validation"
+        with(northbound.validateSwitch(srcSw.dpId)) { validation ->
+            validation.verifyRuleSectionsAreEmpty(["missing", "excess"])
+            validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+        }
+
+        when: "Restore the deleted ISL"
+        antiflap.portUp(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort)
+        Wrappers.wait(WAIT_OFFSET) { assert islUtils.getIslInfo(islToManipulate).get().state == DISCOVERED }
+
+        then: "ISL rule is installed for restored ISL on the both switches"
+        Wrappers.wait(PATH_INSTALLATION_TIME) {
+            [srcSw, dstSw].each { sw ->
+                def islRules = northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
+                    Cookie.isIslVlanEgress(it.cookie)
+                }
+                with(islRules) { rules ->
+                    rules.size() == relatedIsls[sw.dpId].size()
+                    rules*.instructions.goToTable.unique() == [4]  // 4 - egress table id
+                    islRules*.match.inPort.sort() == relatedIsls[sw.dpId]*.srcPort.collect { it.toString() }.sort()
+                }
+            }
+        }
+
+        when: "Update switch properties(multi_table:false) on the dst switch"
+        northbound.updateSwitchProperties(dstSw.dpId, northbound.getSwitchProperties(dstSw.dpId).tap {
+            it.multiTable = false
+        })
+
+        then: "ISL rule is removed on the dst switch"
+        Wrappers.wait(RULES_DELETION_TIME) {
+            assert northbound.getSwitchRules(dstSw.dpId).flowEntries.findAll {
+                Cookie.isIslVlanEgress(it.cookie)
+            }.empty
+        }
+
+        and: "ISL rule is still exist on the src switch"
+        northbound.getSwitchRules(srcSw.dpId).flowEntries.findAll {
+            Cookie.isIslVlanEgress(it.cookie)
+        }.size() == relatedIsls[srcSw.dpId].size()
+
+        and: "Cleanup: revert system to original state"
+        [srcSw, dstSw].each { sw ->
+            northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId)
+                    .tap { it.multiTable = initSwProps[sw.dpId].multiTable })
+            Wrappers.wait(RULES_INSTALLATION_TIME) {
+                assert northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.sort() == sw.defaultCookies.sort()
+            }
+        }
+    }
```

Comment:
We already have a similar test where we delete-rediscover an isl. I'm wondering whether we really want to look at which exactly rules are installed to discover those ISLs. My point is that form functional perspective the fact that isl is successfully rediscovered should be enough.

Reply:
This spec was deleted